concordium-consensus/src/Concordium/GRPC2.hs

```diff
@@ -535,11 +563,714 @@ instance ToProto AccountInfo where
+    ProtoFields.addIdentityProvider .= toProto (Updates.asAddIdentityProvider auth)
+
+instance ToProto Updates.AccessStructure where
+  type Output Updates.AccessStructure = Proto.AccessStructure
+  toProto Updates.AccessStructure{..} = Proto.make $ do
+    ProtoFields.accessPublicKeys .= map toProtoUpdateKeysIndex (Set.toList accessPublicKeys)
+    ProtoFields.accessThreshold .= toProto accessThreshold
+    where toProtoUpdateKeysIndex i = Proto.make $ ProtoFields.value .= fromIntegral i
+
+instance ToProto Updates.UpdatePublicKey where
+  type Output Updates.UpdatePublicKey = Proto.UpdatePublicKey
+  toProto = mkSerialize
+
+instance ToProto Updates.UpdateKeysThreshold where
+  type Output Updates.UpdateKeysThreshold = Proto.UpdateKeysThreshold
+  toProto Updates.UpdateKeysThreshold{..} = Proto.make $ ProtoFields.value .= fromIntegral uktTheThreshold
+
+instance ToProto MintRate where
+  type Output MintRate = Proto.MintRate
+  toProto MintRate{..} = Proto.make $ do
+    ProtoFields.mantissa .= mrMantissa
+    ProtoFields.exponent .= fromIntegral mrExponent
+
+instance ToProto Wasm.WasmVersion where
+  type Output Wasm.WasmVersion = Proto.ContractVersion
+  toProto Wasm.V0 = Proto.V0
+  toProto Wasm.V1 = Proto.V1
+
+instance ToProto Wasm.ContractEvent where
+  type Output Wasm.ContractEvent = Proto.ContractEvent
+  toProto = mkSerialize
+
+instance ToProto CredentialType where
+  type Output CredentialType = Proto.CredentialType
+  toProto Initial = Proto.CREDENTIAL_TYPE_INITIAL
+  toProto Normal = Proto.CREDENTIAL_TYPE_NORMAL
+
+type BakerAddedEvent = (BakerKeysEvent, Amount, Bool)
+
+instance ToProto BakerAddedEvent where
+  type Output BakerAddedEvent = Proto.BakerEvent'BakerAdded
+  toProto (keysEvent, stake, restakeEarnings) = Proto.make $ do
+    ProtoFields.keysEvent .= toProto keysEvent
+    ProtoFields.stake .= toProto stake
+    ProtoFields.restakeEarnings .= restakeEarnings
+
+type BakerKeysEvent = (BakerId, AccountAddress, BakerSignVerifyKey, BakerElectionVerifyKey, BakerAggregationVerifyKey)
+instance ToProto BakerKeysEvent where
+  type Output BakerKeysEvent = Proto.BakerKeysEvent
+  toProto (bakerId, addr, signKey, electionKey, aggregationKey) = Proto.make $ do
+    ProtoFields.bakerId .= toProto bakerId
+    ProtoFields.account .= toProto addr
+    ProtoFields.signKey .= toProto signKey
+    ProtoFields.electionKey .= toProto electionKey
+    ProtoFields.aggregationKey .= toProto aggregationKey
+
+instance ToProto BakerSignVerifyKey where
+  type Output BakerSignVerifyKey = Proto.BakerSignatureVerifyKey
+  toProto = mkSerialize
+
+instance ToProto BakerElectionVerifyKey where
+  type Output BakerElectionVerifyKey = Proto.BakerElectionVerifyKey
+  toProto = mkSerialize
+
+instance ToProto BakerAggregationVerifyKey where
+  type Output BakerAggregationVerifyKey = Proto.BakerAggregationVerifyKey
+  toProto = mkSerialize
+
+instance ToProto Memo where
+  type Output Memo = Proto.Memo
+  toProto = mkSerialize
```

Comment:
Since Memo is already a byte array, would we rather not just return that, rather than serialized with a length upfront?